dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZenodoHelper.java

```diff
@@ -183,6 +189,25 @@ private static void createAliasAndsetUpRelatedIdentifiers(DepositMetadata deposi
     }
 
 
+    /**
+     * extract a digital object identifier (DOI) from a DOI target URL
+     * @param doiUrl digital object identifier
+     * @return the DOI as a string
+     */
+    protected static String extractDoiFromDoiUrl(String doiUrl) {
+        // Remove the 'https://doi.org/' etc. prefix from the DOI
+        // e.g. https://doi.org/10.5072/zenodo.372767
+        String doi = doiUrl;
+        try {
+            URI uri = new URI(doiUrl);
+            doi = StringUtils.stripStart(uri.getPath(), "/");
+
+        } catch (URISyntaxException e) {
+            LOG.error("Could not extract DOI. Error is " + e.getMessage(), e);
```

Comment:
Seems a little odd that this occurrence is serious enough to warrant `LOG.error` but not serious enough to actually do anything and just return the original DOI